### PR TITLE
Generate USWDS theme settings from config tokens

### PR DESCRIPTION
**Pull request by kmonahan**
USWDS settings that should be compiled into `$theme-` variables can be added to the `uswds` section of `config.design-tokens.yml`. The key should be the variable name, minus the `$theme-` prefix.

Example:
```
uswds:
  root-font-size: 16px
```
becomes
```
$theme-root-font-size: 16px;
```
in `_uswds-theme-settings.scss`.

The USWDS tokens can then be referenced by Gesso tokens if needed. For tokens that can be used as is, the token can just be set to `uswds.[token-name]`.

Example:
```
gesso:
  typography:
    base-font-size: uswds.root-font-size
```
ensures that the Gesso base font size used for rem calculations will always match the USWDS root font size (16px).

In places where the USWDS setting is a named design token, the `!uswds` tag can be used to transform the value Gesso uses into a Sass function. Probably makes more sense with an example, so take a look at how gutter-width is handled. Gesso's gutter-width token is set to match the column-gap-desktop value used by USWDS in its theme, which is also changed from the default value to 40px (the default in vanilla Gess). USWDS requires that value to be a spacing token:
```
uswds:
  column-gap-desktop: 5
```
Gesso needs this as `units(5)` since otherwise it won't check USWDS's system properties map to compute the value. To avoid manually keeping those tokens in sync, you can do
```
gesso:
  gutter-width: !uswds
   value: uswds.column-gap-desktop
   fn: 'units'
```
where `value` is the name of one your USWDS tokens in the same config file.

There's also a shorthand method:
```
gesso:
  gutter-width: !uswds uswds.column-gap-desktop
```
The shorthand method defaults to the `units()` Sass function, since that seemed to be the most common use case I was encountering.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@ node_modules
 pattern-lab
 source/_data/design-tokens.artifact.yml
 source/_patterns/00-config/_design-tokens.artifact.scss
+source/_patterns/_uswds-theme-settings.scss
```
Comment (dcmouyard) [inline .gitignore]: How do you feel about changing this filename to `_uswds-theme-settings.artifact.scss`?

**Review by dcmouyard — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (kmonahan) [reply .gitignore]: Good idea, that makes it more consistent with the other generated files. Will update.

Comment (kmonahan) [reply .gitignore]: Updated!

**Review by dcmouyard — APPROVED** [review]
Works great! 👍

**Pull request merged**